cdap-tms/src/main/java/co/cask/cdap/messaging/TopicMetadata.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.messaging;
+
+import co.cask.cdap.proto.id.TopicId;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Represents metadata about a messaging topic.
+ */
+public class TopicMetadata {
+
+  private static final String TTL_KEY = "ttl";
```

Comment:
Since TTL is a required property, the default value will be set by the MessagingService or is it always expected to be provided by the client when creating a topic?